pkg/reconciler/taskrun/resources/apply_test.go

```diff
@@ -577,6 +452,61 @@ func TestApplyParameters(t *testing.T) {
 		})
 	}
 }
+func TestApplyParameters(t *testing.T) {
+	tr := &v1alpha1.TaskRun{
+		Spec: v1alpha1.TaskRunSpec{
+			Inputs: v1alpha1.TaskRunInputs{
+				Params: []v1alpha1.Param{{
+					Name:  "myimage",
+					Value: *builder.ArrayOrString("bar"),
+				}, {
+					Name:  "FOO",
+					Value: *builder.ArrayOrString("world"),
+				}},
+			},
+		},
+	}
+	dp := []v1alpha1.ParamSpec{{
+		Name:    "something",
+		Default: builder.ArrayOrString("mydefault"),
+	}, {
+		Name:    "somethingelse",
+		Default: builder.ArrayOrString(""),
+	}}
+	want := applyMutation(simpleTaskSpec, func(spec *v1alpha1.TaskSpec) {
```

Comment:
`applyMutation` is grooooossss and we should get rid of it. But that's not for this PR.

Reply:
haha its very "clever"